hatice-ecevit-2016400138/assignment2/BinaryTreeNode.java

```diff
@@ -0,0 +1,118 @@
+import java.util.ArrayList;
+public class BinaryTreeNode<T> {
+    private T data;
+    private BinaryTreeNode<T> right;
+    private BinaryTreeNode<T> left;
+
+    public BinaryTreeNode(T data, BinaryTreeNode<T> left, BinaryTreeNode<T> right) {
+        this.data = data;
+        this.left = left;
+        this.right = right;
+    }
+
+    public T getData() {
+        return data;
+    }
+
+    public BinaryTreeNode<T> getRight() {
+        return right;
+    }
+
+    public BinaryTreeNode<T> getLeft() {
+        return left;
+    }
+
+    public void setData(T data) {
+        this.data = data;
+    }
+
+    public void setRight(BinaryTreeNode<T> right) {
+        this.right = right;
+    }
+
+    public void setLeft(BinaryTreeNode<T> left) {
+        this.left = left;
+    }
+
+    /**
+     * @param key key of the node whose ancestors are going to be printed
+     * @return ancestor list of key
+     */
+    ArrayList<T> findAncestors(T key) {
+        // the point at which the key is found
```

Comment:
You have made a decision about what will happen if a node is in the tree twice. It would be good to have that documented in your javadoc.